### PR TITLE
Does maintenance on mechs, adds in the hypergauss.

### PR DESCRIPTION
**Pull request by CRITAWAKETS**
This PR does some maintenance on the modular and even non-modular mech stuff to make it up-to-date.

Non-Modular Stuff Changes:
Mech Zoom is now handled by a variable rather than a hard-coded zoom value. This means you can make a mech zoom out to the entire map if you wish. **(this uses non-modular code)**
The Marauder now starts access-locked, since it's kind of supposed to, given how it has a centcom specops lock on it that just... Isn't normally used.

Modular changes:
A bit of code organization changes to make things properly named and such.
The ER Laser no longer has an invalid negative bare wound bonus value.
The Upgraded Centcom mechs have the appropriate number of equipment slots now.
The Upgraded Centcom mechs now come with tier 4 stock parts pre-equipped.
There are now equipped variants of the upgraded centcom mechs.
The PPC has been rebalanced to be weak against armour, since it's middleground role has been replaced a long time ago with the ER Laser. ER PPC is still strong against armour though.
There is now a new Nanotrasen variant of the Gauss Rifle, which fires _penetrating_ gauss rounds.

### CODE REVIEW

File: maplestation_modules/code/modules/vehicles/mecha/combat/marauder.dm [hunk 1]
```diff
@@ -1,11 +1,17 @@
+/obj/vehicle/sealed/mecha/marauder
+	mecha_flags = ID_LOCK_ON | CAN_STRAFE | IS_ENCLOSED | HAS_LIGHTS | MMI_COMPATIBLE //Upstream fix for the marauder not having an ID lock by default despite being intended to. You love to see it.
```
Comment (MrMelbert) [inline maplestation_modules/code/modules/vehicles/mecha/combat/marauder.dm:2]: How do you reckon it is "intended" to have an ID lock? I know the Marauder specifically has ID lock flipped which makes me think it's "intended" that it *doesn't* have an ID lock (for admin usage?)

File: maplestation_modules/code/modules/vehicles/mecha/combat/marauder.dm [hunk 2]
```diff
@@ -16,24 +22,46 @@
...
+
+/obj/vehicle/sealed/mecha/marauder/upgraded/loaded
+	equip_by_category = list(
+		MECHA_L_ARM = /obj/item/mecha_parts/mecha_equipment/weapon/energy/pulsed_laser/large,
+		MECHA_R_ARM = /obj/item/mecha_parts/mecha_equipment/weapon/energy/er_laser/heavy,
+		MECHA_UTILITY = list(
+			/obj/item/mecha_parts/mecha_equipment/radio,
+			/obj/item/mecha_parts/mecha_equipment/air_tank/full,
+			/obj/item/mecha_parts/mecha_equipment/thrusters/ion,
+			),
```
Comment (MrMelbert) [inline maplestation_modules/code/modules/vehicles/mecha/combat/marauder.dm:49]:
```suggestion
		),
```

File: maplestation_modules/code/modules/vehicles/mecha/combat/marauder.dm [hunk 3]
```diff
@@ -43,12 +71,32 @@
...
+/obj/vehicle/sealed/mecha/marauder/seraph/upgraded/loaded
+	equip_by_category = list(
+		MECHA_L_ARM = /obj/item/mecha_parts/mecha_equipment/weapon/energy/ppc/hellstar,
+		MECHA_R_ARM = /obj/item/mecha_parts/mecha_equipment/weapon/ballistic/missile_rack/ssrm2,
+		MECHA_UTILITY = list(
+			/obj/item/mecha_parts/mecha_equipment/repair_droid,
+			/obj/item/mecha_parts/mecha_equipment/radio,
+			/obj/item/mecha_parts/mecha_equipment/air_tank/full,
+			/obj/item/mecha_parts/mecha_equipment/thrusters/ion,
+			),
```
Comment (MrMelbert) [inline maplestation_modules/code/modules/vehicles/mecha/combat/marauder.dm:99]:
```suggestion
		),
```

File: maplestation_modules/code/modules/vehicles/mecha/combat/marauder.dm (hunk 1 above)
Comment (CRITAWAKETS) [reply maplestation_modules/code/modules/vehicles/mecha/combat/marauder.dm:2]: The mech happens to have a defined access level (Centcom Spec Ops), despite the lack of ID lock.

Also it actually had an ID lock enabled until the Equipment Rework, so it really just seems like this just ended up being forgotten about in the refactor.

Comment (MrMelbert) [reply maplestation_modules/code/modules/vehicles/mecha/combat/marauder.dm:2]: Isn't the access also used by maintenance panel updating or w/e 

Also I would find it very funny if it was "forgotten about" when it was explicitly left out of the flags. Not saying it's impossible mind you, just funny

**Pull request merged**